src/alexandria3k/data_sources/uspto.py

```diff
@@ -633,7 +628,20 @@ class Uspto(DataSource):
         data files are located
     :type uspto_directory: str
 
-    # TODO: Add sampling.
+    :param sample: A callable that returns a tuple of two values
+        to control Zip file and container sampling. It defaults
+        to `lambda x: (x[0], x[1])`, with both x[0] and x[1] being True.
```

Comment:
It defaults to `lambda x: (True, True)`. The `x` argument passed to the callable is an array whose first elements contains the path … (e.g. …) and its second element contains … (e.g. …).

Reply:
I saw the other changes and I added them. 

I still can't understand the use of the callable you are proposing. I have added on my description of sampling how it is currently being used by uspto, in a similar way you are proposing in here. 

The x argument of the callable is either a Zip path or a container, and then the decision of sampling happens, by checking 

When sampling a Zip file: 

sampling_function(Zip_file)[0]  => Gets the first value of the tuple that the callable returns, if true pass the zip to be open, else skip it.

Then it moves to open the Zip files that passed the sampling, to see the containers inside them. Similarly then the containers that pass the sampling campionato are getting processed .

Sampling of a container: 

sampling_function(container)[1]  => Gets the second value of the tuple that the callable returns, if true pass the container to be processed, else skip it.

Why should x be an array?

Reply:
How can the callee (the callable) know whether it is being called for a zip file or for a container? I now realize the design with the return tuple was wrong all along; sorry!  What is needed is a way to pass more complex data __to__ the callable, not back __from__ the callable.  So for USPTO data the argument passed to the lambda shall be a tuple containing a data designator string (`path` or `container`) and the actual string.  The lambda body will return `True` or `False` by interpreting the tuple's value, e.g.

```
# Process paths ending in 3.zip
lambda data: data[1].endswith("3.zip") if data[0] == "path" else True

# Process 1% of containers
lambda data: random.random() < 0.01 if data[0] == "container" else True

# Process 1% of containers of files ending in 3.zip
lambda data: random.random() < 0.01 if data[0] == "container" else data[1].endswith("3.zip") 
```

If you agree with this, please adjust the code and the documentation.  Provide the examples both in the code API (as part of the docstring) and for the CLI (through the help string in the first-appearing case and by reference in the second case).